client/src/components/AuthContext.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useMemo, useState } from 'react';
+import { accessTokenService } from '../services/accessTokenService.js';
+import { authService } from '../services/authService.js';
+
+export const AuthContext = React.createContext({});
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [isChecked, setChecked] = useState(true);
```

Comment:
The initial state for 'isChecked' should likely be 'false' instead of 'true'. This is because 'isChecked' is used to determine if the authentication check has been completed. It should start as 'false' and be set to 'true' once the check is done.